hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+# Copyright 2017 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""A switch driver for Dell S3048-ON running Dell OS 9.
+
+Uses the XML REST API for communicating with the switch.
+"""
+
+import logging
+from lxml import etree
+import re
+import requests
+import schema
+
+from hil.model import db, Switch, SwitchSession
+from hil.errors import BadArgumentError
+from hil.model import BigIntegerType
+
+logger = logging.getLogger(__name__)
+
+CONFIG = 'config-commands'
+SHOW = 'show-command'
+
+
+class DellNOS9(Switch, SwitchSession):
+    """Dell S3048-ON running Dell NOS9"""
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/dellnos9'
+
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(BigIntegerType,
+                   db.ForeignKey('switch.id'), primary_key=True)
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+    interface_type = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'hostname': basestring,
+            'username': basestring,
+            'password': basestring,
+            'interface_type': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    @staticmethod
+    def validate_port_name(port):
+        """Valid port names for this switch are of the form 1/0/1 or 1/2"""
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if not val.match(port):
+            raise BadArgumentError("Invalid port name. Valid port names for "
+                                   "this switch are of the from 1/0/1 or 1/2")
+        return
+
+    def disconnect(self):
+        pass
+
+    def modify_port(self, port, channel, new_network):
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                self._remove_native_vlan(interface)
+            else:
+                self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
```

Comment:
Maybe just call `get_network_allocator().is_legal_channel_for`?

Reply:
This is copy pasta from the brocade code. But yeah, I could change that.

Reply:
In that case, it should probably get changed there as well.

Actually, this whole method feels like boilerplate that should probably get factored out into a super class or something (similarly, the powerconnect and nexus drivers share a lot of structure like this, hence _console.py).